pybop/models/base_model.py

```diff
@@ -104,8 +104,10 @@ def build(
             The initial state of charge to be used in simulations.
         """
         self.dataset = dataset
-        self.parameters = parameters
-        if self.parameters is not None:
+        if parameters is None:
+            self.parameters = Parameters()
+        else:
+            self.parameters = parameters
```

Comment:
With your update to `self.parameters` initialisation above, we should be able simplify to the below.

```suggestion
        if parameters is not None:
            self.parameters = parameters
```